src/java/org/apache/cassandra/db/compaction/CompactionInfo.java

```diff
@@ -132,6 +132,13 @@ public String toString()
         StringBuilder buff = new StringBuilder();
         buff.append(getTaskType());
 
+        if (getTaskId() != null)
+        {
+            buff.append('(')
```

Comment:
@yifan-c Do we want a space before this. Will it be smashed next to the task type?

Reply:
Right now, it would look something like this: `Compaction(58580fc5-0229-4c4d-a57b-af55683d4839)@f2b29d40-eed8-11ea-83a3-67388102292c(mockks, mockcf1, 0/1000)bytes`

It looks like `bytes` is in the wrong place if there's non-null metadata :)

Reply:
Talked on slack. We would like to change the format of `toString` to
```
TaskType(taskId, progress bytes)@tableId(ks, cf)
```

Reply:
cleaner but to keep in-line with the other compaction logs 

````
logger.info("Compacting ({}) {}", taskId, ssTableLoggerMsg);
logger.info(String.format("Compacted (%s) %d sstables to [%s] to level=%d.  %s to %s (~%d%% of original) in %,dms.  Read Throughput = %s, Write Throughput = %s, Row Throughput = ~%,d/s.  %,d total partitions merged to %,d.  Partition merge counts were {%s}"
```

it may be better to only update `org.apache.cassandra.db.compaction.CompactionInterruptedException` since this is what generates the message of `"Compaction interrupted: " + info`, so maybe this should really just be `"Compaction ("+info.getTaskId()+") interrupted: " + info`; this also lowers the changes

Reply:
Including the `compactionId` in the `CompactionInfo` seems reasonable to me. It might benefit other logging parts that I am not aware of. 
If you strongly prefer to only change the exception, I am good with that. 
WDYT?

Reply:
I'm okay with that, but let's also fix the nonsensical formatting for the unit/bytes while we're here either way.